### DISCUSSION TITLE
How to use spacy ner with blank

### DISCUSSION THREAD

**Discussion by eladbitton** [explosion/spaCy#9141] [Help: Coding & Implementations] [answered]
Hi,

I am trying to use the following code:

```
import spacy

nlp = spacy.blank("en")
nlp.add_pipe('ner')
doc = nlp("Abberton Reservoir is a pumped storage freshwater reservoir in eastern England near the Essex coast, with an area of 700 hectares (1,700 acres). Most of its water is pumped in from the River Stour. It is the largest body of freshwater in its county.")


for ent in ents.ents:
    print(ent.text, ent.start_char, ent.end_char, ent.label_)

```

But it throws an exception:

> python3.6/site-packages/spacy/language.py", line 1003, in __call__
>     raise ValueError(Errors.E109.format(name=name)) from e
> ValueError: [E109] Component 'ner' could not be run. Did you forget to call `initialize()`?

Using spacy.load it works well but I want to load only the NER model and I am not sure that load will do that.

How can I load only the NER model, without anything else given the ISO langauge code?

**Comment by polm** [comment, marked as answer]
You can't do that. The NER component needs a tok2vec. 

If you want to use just NER from a pretrained pipeline, then you can do this:

```
import spacy
nlp = spacy.load("en_core_web_sm", disable=["tagger","parser","lemmatizer","attribute_ruler"])
doc = nlp("John Smith is in London.")
for ent in doc.ents:
    print(ent.label_, ent, sep="\t")
```

You have to keep the `tok2vec` component because NER needs it as input. You can't use a blank model because it doesn't have any trained model - if you add NER with `add_pipe` you'd be using a randomly initialized model, which can't give you meaningful output.

**Reply by adrianeboyd** [reply]
Sorry, to jump in here: in `en_core_web_sm` for v3.0.0 and v3.1.0 (model versions, not spacy versions), the `ner` component does not require the `tok2vec` component.

It's not harmful to include it, it just unused/slower as a result. The `ner` component can run on its own with its internal `tok2vec` model, which is separate from the standalone one that is required for the tagger and parser. See: https://spacy.io/models#design
